internal/service/dns_config/api_delegation_resource_test.go

```diff
@@ -0,0 +1,446 @@
+package dns_config_test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+
+	"github.com/infobloxopen/bloxone-go-client/dns_config"
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/acctest"
+)
+
+func TestAccDelegationResource_basic(t *testing.T) {
+	var resourceName = "bloxone_dns_delegation.test"
+	var v dns_config.ConfigDelegation
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccDelegationBasicConfig("test.123.", "123."),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDelegationExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "fqdn", "test.123."),
+					resource.TestCheckResourceAttr(resourceName, "delegation_servers.#", "1"),
+					// Test Read Only fields
+					resource.TestCheckResourceAttrSet(resourceName, "id"),
+					resource.TestCheckResourceAttrSet(resourceName, "protocol_fqdn"),
+					resource.TestCheckResourceAttrSet(resourceName, "parent"),
```

Comment:
you can use `AttrPair` to check if parent ID is the authzone ID that is in the test